File: Data Structures & Algorithms/two-integer-sum-ii/submission-0.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        l, r = 0, len(nums) - 1

        current_sum = nums[l] + nums[r]

        while current_sum != target:
            if current_sum > target:
                r -= 1
            else:
                l += 1
            
            current_sum = nums[l] + nums[r]
            
        return [l + 1, r + 1]

File: Data Structures & Algorithms/combination-target-sum/submission-0.js
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} target
     * @returns {number[][]}
     */
    combinationSum(nums, target) {
        let r = new Array();
        let c = new Array();
        this.backtrack(nums, target, r, c, 0);
        return r;
    }

    backtrack(n, t, r, c, i) {
        if (t === 0) {
            r.push([...c]);
        } else if (t < 0 || i >= n.length) {
            return;
        } else {
            c.push(n[i]);
            this.backtrack(n, t - n[i], r, c, i);

            c.pop();
            this.backtrack(n, t, r, c, i + 1);
        }
    }
}
